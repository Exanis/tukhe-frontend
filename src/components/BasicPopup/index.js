import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import { defineMessages, injectIntl } from 'react-intl';

import Dialog from '@material-ui/core/Dialog';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import MuiDialogActions from '@material-ui/core/DialogActions';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';

const messages = defineMessages({
    close: {
        id: 'composant.BasicPopup.close',
        defaultMessage: 'Close'
    }
});

export const RawDialogTitle = props => {
  const { children, classes, onClose } = props;

  return (
    <MuiDialogTitle disableTypography className={classes.root}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton aria-label={props.intl.formatMessage(messages.close)} className={classes.closeButton} onClick={onClose}>
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
};

export const DialogTitle = injectIntl(withStyles(theme => ({
  root: {
    borderBottom: `1px solid ${theme.palette.divider}`,
    margin: 0,
    padding: theme.spacing.unit * 2,
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing.unit,
    top: theme.spacing.unit,
    color: theme.palette.grey[500],
  },
}))(RawDialogTitle));

DialogTitle.propTypes = {
    onClose: PropTypes.func
};

export const DialogContent = withStyles(theme => ({
  root: {
    margin: 0,
    padding: theme.spacing.unit * 2,
  },
}))(MuiDialogContent);

export const DialogActions = withStyles(theme => ({
  root: {
    borderTop: `1px solid ${theme.palette.divider}`,
    margin: 0,
    padding: theme.spacing.unit,
  },
}))(MuiDialogActions);

class BasicPopup extends Component {
    renderTitle = () => {
        if (!this.props.title)
            return null;
        return <DialogTitle onClose={this.props.onClose}>
            {this.props.title}
        </DialogTitle>;
    };

    renderActions = () => {
        if (!this.props.actions)
            return null;
        return <DialogActions>
            {this.props.actions}
        </DialogActions>;
    };

    render() {
        return <Dialog
            {...this.props}
            >
            {this.renderTitle()}
            <DialogContent>
                {this.props.children}
            </DialogContent>
            {this.renderActions()}
        </Dialog>;
    }
}

BasicPopup.propTypes = {
    title: PropTypes.string,
    onClose: PropTypes.func,
    actions: PropTypes.node,
};

export default BasicPopup;
