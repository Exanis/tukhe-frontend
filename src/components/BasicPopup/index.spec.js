import React from 'react';
import { IntlProvider } from 'react-intl';
import BasicPopup, { RawDialogTitle, DialogTitle, DialogContent, DialogActions } from './index';
import { shallow, render } from 'enzyme';

it('Should render correctly with all props', () => {
    const props = {
        title: 'Test title',
        onClose: jest.fn(),
        actions: [<div key="1">Test div</div>],
        additionalProp: {}
    };
    const wrapper = shallow(<BasicPopup {...props} />);

    expect(wrapper).toMatchSnapshot();
});

it('Should render correctly without title', () => {
    const props = {
        actions: [<div key="1">Test div</div>],
    };
    const wrapper = shallow(<BasicPopup {...props} />);

    expect(wrapper).toMatchSnapshot();
});

it('Should render correctly without actions', () => {
    const props = {
        title: 'Test title',
        onClose: jest.fn(),
    };
    const wrapper = shallow(<BasicPopup {...props} />);

    expect(wrapper).toMatchSnapshot();
});

it('Should render title correctly with close button', () => {
    const props = {
        onClose: jest.fn(),
        classes: {},
        intl: {
            formatMessage: jest.fn()
        }
    };
    const wrapper = render(<RawDialogTitle {...props}><div>Test</div></RawDialogTitle>);

    expect(wrapper).toMatchSnapshot();
});

it('Should render title correctly without close button', () => {
    const props = {
        classes: {},
        intl: {
            formatMessage: jest.fn()
        }
    };
    const wrapper = render(<RawDialogTitle {...props}><div>Test</div></RawDialogTitle>);

    expect(wrapper).toMatchSnapshot();
});

it('Should render popup title as expected', () => {
    const wrapperTitle = render(<IntlProvider locale="en"><DialogTitle /></IntlProvider>);

    expect(wrapperTitle).toMatchSnapshot();
});

it('Should render popup content as expected', () => {
    const wrapperContent = render(<IntlProvider locale="en"><DialogContent /></IntlProvider>);

    expect(wrapperContent).toMatchSnapshot();
});

it('Should render popup actions as expected', () => {
    const wrapperActions = render(<IntlProvider locale="en"><DialogActions /></IntlProvider>);

    expect(wrapperActions).toMatchSnapshot();
});
