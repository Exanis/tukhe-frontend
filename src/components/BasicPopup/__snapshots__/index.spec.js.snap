// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly with all props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BasicPopup
    actions={
      Array [
        <div>
          Test div
        </div>,
      ]
    }
    additionalProp={Object {}}
    onClose={[MockFunction]}
    title="Test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "actions": Array [
        <div>
          Test div
        </div>,
      ],
      "additionalProp": Object {},
      "children": Array [
        <InjectIntl(WithStyles(RawDialogTitle))
          onClose={[MockFunction]}
        >
          Test title
        </InjectIntl(WithStyles(RawDialogTitle))>,
        <WithStyles(WithStyles(DialogContent)) />,
        <WithStyles(WithStyles(DialogActions))>
          <div>
            Test div
          </div>
        </WithStyles(WithStyles(DialogActions))>,
      ],
      "onClose": [MockFunction],
      "title": "Test title",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Test title",
          "onClose": [MockFunction],
        },
        "ref": null,
        "rendered": "Test title",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              Test div
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Test div",
            },
            "ref": null,
            "rendered": "Test div",
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <div>
            Test div
          </div>,
        ],
        "additionalProp": Object {},
        "children": Array [
          <InjectIntl(WithStyles(RawDialogTitle))
            onClose={[MockFunction]}
          >
            Test title
          </InjectIntl(WithStyles(RawDialogTitle))>,
          <WithStyles(WithStyles(DialogContent)) />,
          <WithStyles(WithStyles(DialogActions))>
            <div>
              Test div
            </div>
          </WithStyles(WithStyles(DialogActions))>,
        ],
        "onClose": [MockFunction],
        "title": "Test title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test title",
            "onClose": [MockFunction],
          },
          "ref": null,
          "rendered": "Test title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                Test div
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Test div",
              },
              "ref": null,
              "rendered": "Test div",
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Should render correctly without actions 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BasicPopup
    onClose={[MockFunction]}
    title="Test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <InjectIntl(WithStyles(RawDialogTitle))
          onClose={[MockFunction]}
        >
          Test title
        </InjectIntl(WithStyles(RawDialogTitle))>,
        <WithStyles(WithStyles(DialogContent)) />,
        null,
      ],
      "onClose": [MockFunction],
      "title": "Test title",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Test title",
          "onClose": [MockFunction],
        },
        "ref": null,
        "rendered": "Test title",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <InjectIntl(WithStyles(RawDialogTitle))
            onClose={[MockFunction]}
          >
            Test title
          </InjectIntl(WithStyles(RawDialogTitle))>,
          <WithStyles(WithStyles(DialogContent)) />,
          null,
        ],
        "onClose": [MockFunction],
        "title": "Test title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test title",
            "onClose": [MockFunction],
          },
          "ref": null,
          "rendered": "Test title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Should render correctly without title 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BasicPopup
    actions={
      Array [
        <div>
          Test div
        </div>,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "actions": Array [
        <div>
          Test div
        </div>,
      ],
      "children": Array [
        null,
        <WithStyles(WithStyles(DialogContent)) />,
        <WithStyles(WithStyles(DialogActions))>
          <div>
            Test div
          </div>
        </WithStyles(WithStyles(DialogActions))>,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              Test div
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Test div",
            },
            "ref": null,
            "rendered": "Test div",
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <div>
            Test div
          </div>,
        ],
        "children": Array [
          null,
          <WithStyles(WithStyles(DialogContent)) />,
          <WithStyles(WithStyles(DialogActions))>
            <div>
              Test div
            </div>
          </WithStyles(WithStyles(DialogActions))>,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                Test div
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Test div",
              },
              "ref": null,
              "rendered": "Test div",
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Should render popup actions as expected 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiDialogActions-root-61 WithStyles-DialogActions--root-60",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Should render popup content as expected 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiDialogContent-root-59 WithStyles-DialogContent--root-58",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Should render popup title as expected 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiDialogTitle-root-1 RawDialogTitle-root-56",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiTypography-root-2 MuiTypography-h6-19",
        },
        "children": Array [],
        "name": "h6",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Should render title correctly with close button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiDialogTitle-root-1",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiTypography-root-2 MuiTypography-h6-19",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Test",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "h6",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "MuiButtonBase-root-44 MuiIconButton-root-38",
            "tabindex": "0",
            "type": "button",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "MuiIconButton-label-43",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "MuiSvgIcon-root-47",
                    "focusable": "false",
                    "role": "presentation",
                    "viewBox": "0 0 24 24",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": Object {
                        "attribs": Object {
                          "d": "M0 0h24v24H0z",
                          "fill": "none",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                          "fill": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                          "fill": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "d": "M0 0h24v24H0z",
                        "fill": "none",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "d": "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                        "fill": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                        "fill": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "focusable": undefined,
                    "role": undefined,
                    "viewBox": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "focusable": undefined,
                    "role": undefined,
                    "viewBox": undefined,
                  },
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "tabindex": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "tabindex": undefined,
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "MuiButtonBase-root-44 MuiIconButton-root-38",
          "tabindex": "0",
          "type": "button",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MuiIconButton-label-43",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "MuiSvgIcon-root-47",
                  "focusable": "false",
                  "role": "presentation",
                  "viewBox": "0 0 24 24",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": Object {
                      "attribs": Object {
                        "d": "M0 0h24v24H0z",
                        "fill": "none",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                        "fill": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                        "fill": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "d": "M0 0h24v24H0z",
                      "fill": "none",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "d": "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                      "fill": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                      "fill": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "focusable": undefined,
                  "role": undefined,
                  "viewBox": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "focusable": undefined,
                  "role": undefined,
                  "viewBox": undefined,
                },
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "MuiTypography-root-2 MuiTypography-h6-19",
          },
          "children": Array [
            Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "Test",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "h6",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "tabindex": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "tabindex": undefined,
          "type": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Should render title correctly without close button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiDialogTitle-root-1",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiTypography-root-2 MuiTypography-h6-19",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Test",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "h6",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
